src/main/java/org/traccar/protocol/WatchProtocolDecoder.java

```diff
@@ -164,8 +164,11 @@ private Position decodePosition(DeviceSession deviceSession, String data) {
 
             for (int i = 0; i < wifiCount; i++) {
                 index += 1; // wifi name
-                network.addWifiAccessPoint(WifiAccessPoint.from(
-                        values[index++], Integer.parseInt(values[index++])));
+                String macAddress = values[index++];
+                String rssi = values[index++];
+               if (!macAddress.isEmpty() && !rssi.isEmpty()) {
+                   network.addWifiAccessPoint(WifiAccessPoint.from(macAddress, Integer.parseInt(rssi)));
+               }
```

Comment:
This part doesn't make sense to me. I also can't find anything about it in the documentation you provided. Can you please point to the page in the doc. Also, please attache the documentation to this ticket.

Reply:
The documentation is now attached.

The reason for this conditional check is not something to be mentioned from new documentation, but I added this because some tests are failing when we move timing advance into the inner block.

For example, bellow test at line 33 in WatchProtocolDecoderTest.java fails:
````
verifyPosition(decoder, buffer(
                "[SG*9059056143*0053*UD,251021,223408,A,41.46500,N,081.53128,W,0.926,000,0,00,70,70,0,50,00000000,0,1,,,,00]"));
````
I see in this piece of code, there is no base stations information and no WiFi information reported, therefore parsing empty string to integer (signal strength) making the problem and producing the error.

Reply:
It is according to the documentations for watch protocol. I read the documentations which are uploaded in Traccar website [here](https://www.traccar.org/protocols/) and the documentation which I uploaded to this ticket. 

The documentation are stating that, right after the status field:
1. Directly after status, there is data for base stations starting with base stations count, count will be greater than zero if base station information is available. But it will be zero if not available. 
2. And in two other documentations (documentation attached to this ticket at page 4 starting line 6 and documentation on Traccar [比电协议V1.8（20160926）.doc](https://www.traccar.org/protocol/5093-watch/%E6%AF%94%E7%94%B5%E5%8D%8F%E8%AE%AEV1.8%EF%BC%8820160926%EF%BC%89.doc) , there is WiFi data directly after base station information. The first integer will state the count for WiFi stations which are reported along with. 

So this way, there is a possible case that the WiFi count field has a value greater than zero but there is no WiFi stations attached or device has failed to transmit the WiFi station data (mac address and rssi). 

So in bellow case, I see the same as above, there is 0 base stations, but the WiFi count field is 1 and there is no WiFi stations attached.

```
verifyPosition(decoder, buffer(
                "[SG*9059056143*0053*UD,251021,223408,A,41.46500,N,081.53128,W,0.926,000,0,00,70,70,0,50,00000000,0,1,,,,00]"));
```

Reply:
What about this test case:
```
[SG*9051004074*0058*AL,120117,145602,V,40.058413,N,76.336618,W,11.519,188,99,00,01,80,0,50,00000000,0,1,0,0,,10]
```

Reply:
The test also pass with it, but a zero mac address withe zero rssi will be added to the position. We can avoid this by a conditional check of mac address with zero. I mean if there is a value of zero for mac address then we do not need to add it.

But the last double in these messages, the documentation says it is accuracy, but as I know, with current implementation of watch protocol and Traccar, the accuracy comes from geolocation providers. If this is true, then this last double can also be ignored.

Reply:
I think at the very least we need to check for those invalid values.

Reply:
With a1bf85f I added a check to exclude mac address if it is zero.